src/Types/Input/InputObjectType.hack

```diff
@@ -0,0 +1,37 @@
+namespace Slack\GraphQL\Types;
+
+use namespace Facebook\{TypeCoerce, TypeAssert};
+
+<<__ConsistentConstruct>>
+abstract class InputObjectType<TCoerced as nonnull> extends InputType<TCoerced> {
+
+    abstract const string NAME;
+    abstract public function coerceValue(mixed $value): TCoerced;
+    abstract public function assertValidVariableValue(mixed $value): TCoerced;
+
+    <<__Override>>
+    final public function getName(): string {
+        return static::NAME;
+    }
+
+    <<__Override>>
+    final public function coerceNonVariableNode(
+        \Graphpinator\Parser\Value\Value $node,
+        dict<string, mixed> $variable_values,
+    ): TCoerced {
+        return $this->coerceValue($node->getRawValue());
```

Comment:
I think this won't work if there are variable references nested inside the input object, but we can fix that later. I have an idea for what might be a good implementation so I could give it a try after you merge this.

```GraphQL
mutation CreateUser($name: String!) {
    createUser(input: {name: $name}) {
        id
        is_active
        name
        team {
            name
        }
    }
}
```